src/Hedgehog/Gen.fs

```diff
@@ -2,18 +2,36 @@
 
 open System
 
+type GenConfig<'a> = private {
+    Formatter : 'a -> string
+}
+
 /// A generator for values and shrink trees of type 'a.
 [<Struct>]
-type Gen<'a> =
-    | Gen of Random<Tree<'a>>
+type Gen<'a> = private {
+    Random : Random<Tree<'a>>
+    Config : GenConfig<'a>
```

Comment:
nit: I would like the order of these fields and the ones for `Property<>` to be the same

Reply:
Done